scripts/content-repo-preview/build.ts

```diff
@@ -0,0 +1,99 @@
+import { promisify } from 'util'
+import { execFileSync } from 'child_process'
+import path from 'path'
+import fs from 'fs'
+
+async function main() {
+  // imports
+  const execFile = promisify((await import('child_process')).execFile)
+
+  // TODO: check for necessary env vars?
+
+  const repo = process.env.REPO
+
+  // our CWD
+  const cwd = process.cwd()
+  const globalCSSFile = path.join(cwd, 'src', 'pages', 'style.css')
+
+  /**
+   * Check for a .next folder, if found copy it back into our website-preview dir
+   * This should allow us to take advantage of Vercel's build cache
+   */
+  if (fs.existsSync(path.join(cwd, '..', '.next'))) {
+    console.log('.next folder found, moving into website-preview...')
+    await execFile('mv', ['../.next', './.next'])
+
+    if (fs.existsSync(path.join(cwd, '.next', 'cache', 'node_modules'))) {
+      console.log('Found cached node_modules, moving...')
+      await execFile('mv', ['./.next/cache/node_modules', './node_modules'])
+    }
+  }
+
+  // copy public files
+  console.log('📝 copying files in the public folder')
+  await execFile('cp', ['-R', './public', '../'])
+
+  /**
+   * exclude any imports in the global CSS file which rely on other products
+   */
+  const globalCSSFileContents = await fs.promises.readFile(
+    globalCSSFile,
+    'utf-8'
+  )
+
+  const newContents = globalCSSFileContents
+    .split('\n')
+    .map((line) => {
+      // comment out lines which references paths we will be removing
+      if (
+        !line.startsWith('/*') &&
+        line.includes('_proxied-dot-io') &&
+        !line.includes(repo)
+      ) {
+        return `/* ${line} */`
+      }
+      return line
+    })
+    .join('\n')
+
+  console.log(`🧹 removing global CSS references for other products`)
+  await fs.promises.writeFile(globalCSSFile, newContents)
+
+  /**
+   * Remove dirs in `src/pages` which are not associated with the product
+   */
+  const proxiedIoPagesDir = path.join(cwd, 'src', 'pages', '_proxied-dot-io')
+
+  const proxiedIoDirs = (
+    await fs.promises.readdir(proxiedIoPagesDir, { withFileTypes: true })
+  ).filter((ent) => ent.isDirectory)
+
+  for (const dir of proxiedIoDirs) {
+    if (!dir.name.includes(repo)) {
+      console.log(`🧹 removing pages for ${dir.name}`)
+      await fs.promises.rm(path.join(proxiedIoPagesDir, dir.name), {
+        recursive: true,
+      })
+    }
+  }
+
+  /** Install deps */
+  console.log('📦 Installing dependencies')
+  execFileSync('npm', ['install', '--include=dev'], { stdio: 'inherit' })
```

Comment:
using `npm install` instead of `npm ci` here as `ci` will actually remove `node_modules` if it already exists.

Reply:
TIL `--include=dev` 👍

I could only find sparse docs on `--include` (under [omit](https://docs.npmjs.com/cli/v7/commands/npm-install#omit)). Is this ~ equivalent to `--production=false`, in the sense that it forces devDepencies even if `NODE_ENV == "production"`?

Reply:
I think it would be equivalent yeah. I found it in a SO answer I think. Happy to switch to --production=false if we think that's clearer!